### PR TITLE
Fix function suggestions with invokable parameter

### PR DESCRIPTION
**Pull request by nadeeshaan**
- Remove invalidated Language constructs
- Fix function suggestion issue when the function takes an invokable symbol as a parameter
- Remove unwanted symbols items from the visible symbols list
- Enable test cases for latest function definition scope resolving

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/AbstractItemResolver.java [hunk 1]
```diff
@@ -189,19 +191,24 @@ private FunctionSignature getFunctionSignature(BInvokableSymbol bInvokableSymbol
...
-                typeName = tSymbol.getName().getValue();
-            } else {
-                typeName = nameComps.get(nameComps.size() - 1).getValue() + UtilSymbolKeys.PKG_DELIMITER_KEYWORD
-                        + tSymbol.getName().getValue();
+                BTypeSymbol tSymbol;
+                if (paramType instanceof BArrayType) {
+                    tSymbol = ((BArrayType) paramType).eType.tsymbol;
+                } else {
+                    tSymbol = paramType.tsymbol;
+                }
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/AbstractItemResolver.java]: can replace above with following

```
BTypeSymbol tSymbol = paramType instanceof BArrayType
? ((BArrayType) paramType).eType.tsymbol : paramType.tsymbol;
```

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/AbstractItemResolver.java [hunk 2]
```diff
@@ -358,4 +365,25 @@ protected void populateBasicTypes(List<CompletionItem> completionItems, List<Sym
...
+    /**
+     * Remove the invalid symbols such as anon types, injected packages and invokable symbols without receiver.
+     * @param symbolInfoList    Symbol info list to be filtered
+     * @return {@link List}     List of filtered symbols
+     */
+    protected List<SymbolInfo> removeInvalidStatementScopeSymbols(List<SymbolInfo> symbolInfoList) {
+        // We need to remove the functions having a receiver symbol and the bTypes of the following
+        // ballerina.coordinator, ballerina.runtime, and anonStructs
+        ArrayList<String> invalidPkgs = new ArrayList<>(Arrays.asList("ballerina.runtime",
+                "ballerina.transactions"));
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/AbstractItemResolver.java]: package name should be change to "org/pkg_name" or just "pkg_name"

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/definitions/FunctionDefinition.java [hunk 3]
```diff
@@ -30,21 +30,21 @@
...
-                {"enumSuggestVarDef1.json", "function"},
-                {"enumSuggestVarDef2.json", "function"},
-                {"structFields.json", "function"},
-                {"structBoundFunctionsAndFields.json", "function"},
+//                {"enumSuggestAssignment1.json", "function"},
+//                {"enumSuggestAssignment2.json", "function"},
+//                {"enumSuggestVarDef1.json", "function"},
+//                {"enumSuggestVarDef2.json", "function"},
+//                {"structFields.json", "function"},
+//                {"structBoundFunctionsAndFields.json", "function"},
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/definitions/FunctionDefinition.java]: can we remove these as they are no longer valid?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/AbstractItemResolver.java (hunk 1 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/AbstractItemResolver.java]: Fixed

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/AbstractItemResolver.java (hunk 2 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/resolvers/AbstractItemResolver.java]: Fixed

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/definitions/FunctionDefinition.java (hunk 3 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/definitions/FunctionDefinition.java]: Will keep until the latest syntax related tests added

**Pull request merged**